packages/@aws-cdk/aws-route53/lib/hosted-zone.ts

```diff
@@ -222,14 +222,16 @@ export class PublicHostedZone extends HostedZone implements IPublicHostedZone {
   /**
    * Import a Route 53 public hosted zone defined either outside the CDK, or in a different CDK stack
    *
+   * Use when hosted zone ID is known. Hosted zone name becomes unavailable through this query.
```

Comment:
```suggestion
   * Use when hosted zone ID is known. If a HostedZone is imported with this method the zoneName cannot be referenced. If the zoneName is needed then the HostedZone should be imported with fromPublicHostedZoneAttributes.
```

Maybe something like this (for the other instances as well)? I'll leave it up to you.